pkg/model/job.go

```diff
@@ -275,8 +278,13 @@ func (s *Spec) GetTimeout() time.Duration {
 
 // Return pointers to all the storage specs in the spec.
 func (s *Spec) AllStorageSpecs() []*StorageSpec {
-	storages := []*StorageSpec{
-		&s.Wasm.EntryModule,
+	storages := []*StorageSpec{}
+	if s.EngineSpec.Type == EngineTypeWasm {
+		wasmEngine, err := WasmEngineFromEngineSpec(s.EngineSpec)
+		if err != nil {
+			panic(err)
+		}
+		storages = append(storages, &wasmEngine.EntryModule)
```

Comment:
bubbling up the error from this method will be a moderately size code change, if we feel strongly that panicking is incorrect here I will make said change.

Reply:
Not a fan of panics, and AllStorageSpecs() isn't called from too many places, _but_ the problem is that places like `NewStoragesNodeRanker` will need to handle the error, and don't currently return an error, which means its use in the ranker chain will need to change to accommodate it.